GsaGH/Components/2_Geometry/ShowId.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Grasshopper.Kernel;
+using Grasshopper.Kernel.Types;
+using GsaGH.Parameters;
+using GsaGH.Helpers.GH;
+using OasysGH;
+using OasysGH.Components;
+using OasysUnits.Units;
+using Rhino.Geometry;
+using OasysUnits;
+using GsaAPI;
+using Rhino.Collections;
+using Grasshopper.Kernel.Data;
+
+namespace GsaGH.Components
+{
+  /// <summary>
+  /// Component to edit a Node
+  /// </summary>
+  public class ShowId : GH_OasysComponent, IGH_PreviewObject
+  {
+    private List<Point3d> _pts;
+    private List<string> _txts;
+
+    #region Name and Ribbon Layout
+    public override Guid ComponentGuid => new Guid("e01fde68-b591-4ada-b590-9506fc962114");
+    public override GH_Exposure Exposure => GH_Exposure.quinary | GH_Exposure.obscure;
+    public override OasysPluginInfo PluginInfo => GsaGH.PluginInfo.Instance;
+    protected override System.Drawing.Bitmap Icon => GsaGH.Properties.Resources.ShowID;
+
+    public ShowId() : base("ShowID",
+      "ID",
+      "Show the ID of a Node, Element, Member geometry or Result parameters",
+      CategoryName.Name(),
+      SubCategoryName.Cat2())
+    { }
+    #endregion
+
+    #region Input and output
+    protected override void RegisterInputParams(GH_InputParamManager pManager)
+    {
+      pManager.AddGenericParameter("Node/Element/Member/Result", "Geo", "Node, Element, Member or Point/Line/Mesh result to get ID for.", GH_ParamAccess.tree);
+    }
+
+    protected override void RegisterOutputParams(GH_OutputParamManager pManager)
+    {
+      pManager.AddPointParameter("Position", "P", "The (centre/mid) location(s) of the object(s)", GH_ParamAccess.tree);
+      pManager.HideParameter(0);
+      pManager.AddIntegerParameter("Index", "ID", "The objects ID(s)", GH_ParamAccess.tree);
+    }
+    #endregion
+
+    protected override void SolveInstance(IGH_DataAccess DA)
```

Comment:
da is an argument, should be lowerCase ;)

Reply:
yep, it is DA everywhere though?

Reply:
yes but still it's broke google style guide